File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int l=0,r=nums.length-1,m=0;
        while(l<r)
        {
            m=(l+r)/2;
            if(nums[m]>nums[r])
            {
                l=m+1;
            }
            else 
            {
                r=m;
            }
        }
        
        return nums[l];
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        p,z=1,0
        l=[0]*len(nums)
        for i in range(len(nums)):
            if nums[i]==0:
                z+=1
            else:
                p*=nums[i]
        if z>1:
            return l
        for i in range(len(nums)):
            if z==1 and nums[i]==0:
                    l[i]=p
            elif z==1:
                l[i]=0
            else:
                l[i]=p//nums[i]
        return l    
        